### PR TITLE
Complete Mongoose setup

### LINKED ISSUE #12: Setup mongodb

### PR DESCRIPTION
**Pull request by rollaball**
add schema got users, messages and channels

#### What does this PR do?
(A brief explanation synthesizing the feature, bug or fix.)
- adds mongoose setup
- adds schema for channels, users and messages

## Checklist

(Go over all the following points, and put an `x` in all the boxes that apply.)
(If you're unsure about any of these, don't hesitate to ask. We're here to help!)

- [x] My code follows the code style of this project.
- [x] Make sure that you don't have **[WIP]** if you have completed your PR.
- [x] **I am choosing the correct base branch to merge**. 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] And more importantly, I've had fun! :beer:

#### Where should the reviewer start?
(Explain where the reviewer should start in order to review the whole addition or subtractions.)
:checkered_flag:

#### How should this be manually tested?
(List of steps to reproduce, corroborate or tests to run. Write this section clear enough so that external users can also follow it and test the fix.)
#### Any background context you want to provide?
(Any information regarding the PR that the reviewer should know.)
:construction:

#### What are the relevant tickets?
(Link to issues, related PR, JIRA issues, etc.)
Closes #12 

#### Screenshots
(Screenshots of the feature if available.)
:camera:

#### Questions
(If available, to another developer or reviewer.)
:question:

### CODE REVIEW

**Comment by abhi18av** [comment]
@rollaball , please rebase this PR and get it reviewed. The `circleci` setup is completed now.

**Review by nimish-gupta — APPROVED** [review]
LGTM, just a doubt do you guys not need timestamp?

File: server/src/Schemas/messages.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import mongoose from 'mongoose';
+
+const Schema = mongoose.Schema;
+const ObjectId = Schema.ObjectId;
+
+const MessageSchema = new Schema({
+  message: String,
+  read: Boolean,
+  timestamp: Date,
```
Comment (rollaball) [inline server/src/Schemas/messages.js:9]: Using the date as a timestamp.
Are you talking about this?
@nimish-gupta

Comment (nimish-gupta) [reply server/src/Schemas/messages.js:9]: so there is a default timestamp option in mongoose, you can use that.

**Pull request merged**